### ISSUE TITLE
can't import projects on ipad from android???

### ISSUE THREAD

**Issue by ksdavidc** [LLK/scratchjr#268]
Installed latest version on ipad and android. Ipad received sjr file created from Android, that imports fine on android (phone, mac via bluestacks, windows via nox). Sent sjr various ways: shared file via Line, and email. When email opened or file opened, no option to open with scratch jr. appears. Tried saving and opening it to 3 different file managers: ES FIle explorer, FE FIle Explorer and File manager. None of them able to successfully open the file. On the other hand, files created on ipad open fine in android. Tried sending to Google drive. No go. What am I missing?

**Comment by ksdavidc** [comment]
Version 12.4. on ipad.

**Comment by chrisgarrity** [comment]
@ksdavidc can you confirm that you have the latest version (1.2.9) on both platforms - it was in the process of rolling out this past weekend. 
Also, if you email a project from the iPad, will the iPad open that file? Any additional details you can provide will help us to reproduce and narrow down the problem.

**Comment by ksdavidc** [comment]
I tried it on an tablet on which the app was downloaded from the app store for the first time yesterday evening. We tried via email, but the problem is that scratch never becomes a choice for opening the file, and using file explorers doesn't work either. The files are saved. I also tried via the links at the bootup curriculum.

I don't actually have an ipad (students do..), but was working with a borrowed one, which I have returned, so I can't confirm the actual version number installed, but I checked the app store right now and it says 1.2.9. It was on the japanese app store last night.

Are you telling me it works on your ipads? Can you send me screenshots of what I am supposed to see and the steps involved?

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by ksdavidc** [comment]
P.S. Forgot to say thanks for the reply!! If you, we, could get this working it would be a real blessing for us. I am teaching Scratchjr to a bunch of kids here in japan and being able to send and receive projects would be awesome!!! I spent hours trying to do it on my mac/pc via both the community and emulator versions thinking I was doing something wrong, and then you came along with the update, and I thought I was in Heaven. Unofrtunately all my students us ipads. I feel so close, yet so far.... Why no file open/save function? Isn't that the way these things usually go?

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by chrisgarrity** [comment]
Here are the steps I followed on a Samsung tablet running Android 5.1.1 and iPad running iOS 12.1.4.

On Android tablet running ScratchJr 1.2.9:
1. Share by email
2. Select Gmail from the options
3. See that file is attached with name <project>.sjr

On iPad:
1. ScratchJr is already open
2. Open email with attachment
3. Tap to download the file
4. Once the file is downloaded, tap the attachment again
5. Select 'Copy to ScratchJr' from the list of options.

In google drive on iOS:
1. browse to the project
2. Drive says 'unsupported file type'
3. tap the 3-dot menu, and select 'Open in'
4. select 'Copy to ScratchJr'

Earlier versions of ScratchJr on Android had a bug where they set the mimetype incorrectly. If you are only seeing the option to open the file in email and file apps, it may have been sent from an earlier version.

**Comment by ksdavidc** [comment]
Thanks. The part that didn't seem to be working is that there was no "copy to scratchhr" part. Android works. I will have to check again...

(Alan...this is from the developer, can you check on your ipad. ?)

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by ksdavidc** [comment]
Thank you very much.
I’ve tried as described and even with ScratchJr opened, I do not see the option “Copy to ScratchJr”

**Comment by yueyuzhao** [comment]
Hi @ksdavidc , could you please send your project to [donald@drmer.net](mailto:donald@drmer.net) or upload your project here so that I can help to diagnose where the problem is.

**Comment by ksdavidc** [comment]
sure. Attached are 3 from different sources, including ipad

**Comment by ksdavidc** [comment]
![unnamed-2](https://user-images.githubusercontent.com/62230004/80431769-4d33de80-892d-11ea-9ae0-31db8976ef68.jpg)
![unnamed](https://user-images.githubusercontent.com/62230004/80431773-502ecf00-892d-11ea-9c78-65650fb8d0c4.jpg)
![unnamed-1](https://user-images.githubusercontent.com/62230004/80431774-515ffc00-892d-11ea-9004-9acab89e7b6f.jpg)
![unnamed-3](https://user-images.githubusercontent.com/62230004/80431776-515ffc00-892d-11ea-9334-c0fa2c35a803.jpg)

**Comment by ksdavidc** [comment]
misclick, sorry.

**Comment by ksdavidc** [comment]
Here are some samples.

[Samples.zip](https://github.com/LLK/scratchjr/files/4542577/Samples.zip)

**Comment by yueyuzhao** [comment]
Hi @ksdavidc , this is weird. Old version of ScratchJr are able to open those projects, but the latest version (1.2.9) is unable to do so. The bad news is after I pull the latest code and run it on my iPad(12.4.3), I can open these projects. Further investigation needed.

**Comment by yueyuzhao** [comment]
After a comparison of the `Info.plist`, I found a small difference in the production file of `CFBundleDocumentTypes` field:
```
<key>LSHandlerRank</key>
<string>Owner</string>
```
And adding this to Free-Info.plist, there we go: the app is not on the share menu any more.

**Comment by yueyuzhao** [comment]
> After a comparison of the `Info.plist`, I found a small difference in the production file of `CFBundleDocumentTypes` field:
> 
> ```
> <key>LSHandlerRank</key>
> <string>Owner</string>
> ```
> 
> And adding this to Free-Info.plist, there we go: the app is not on the share menu any more.

Oops, sorry I was wrong. The reason is this field `LSItemContentTypes` below:
```
<key>LSItemContentTypes</key>
<array>
    <string>org.scratchfoundation.sjr</string>
</array>
```
after change the value to `com.playfulinvention.sjr`, every thing works fine.

**Comment by chrisgarrity** [comment]
@ksdavidc @yueyuzhao I have a new build (1.2.10) that imports those projects (and earlier saved sjr files) on my test iPad. Unfortunately with the lockdown, I only have one iPad to test on. If you'd like to help test it, send info@scratchjr.org a message that includes the email associated with your iPad. I can invite you to try it out on testflight. Thanks.

**Comment by chocochocosensei** [comment]
@chrisgarrity I will tell that to the person with the iPad and you should get an email. THANK YOU SO MUCH!!!!!!

**Comment by chocochocosensei** [comment]
@yueyuzhao  Thank you so much for troubleshooting this. working on testing the new build, fingers crossed.

**Comment by yueyuzhao** [comment]
@chocochocosensei You're welcome :)

**Comment by ksdavidc** [comment]
@chrisgarrity @yueyuzhao @BryceLTaylor I am sorry to say this has not been fully fixed. I am still getting this error. I posted it in [issues](https://github.com/LLK/scratchjr/issues/394), but the post was summarily ignored (not triaged), so I am trying this route. 

In addition to what is in the post, I should add that I tried several methods of transferring the file, aka via airdrop and via emails, to rule out it being related to that. 

Also, to clarify, the 2nd ipad mentioned could receive and open the files, but when sent from this ipad, they also failed. 

Also note inclusion of 2nd problem related to the home screen. Not sure if related, but it is quite an annoyance. I thought it was just a bluestacks thing, but I confirmed two days ago that it is happening on my student's native Android machine, just not mine! Weird.

For ease of reference, here is the issue post:

I am still getting this problem. It is quite mystifying. I can open some files created by the same ipad, but not others (probably why it appeared to have been fixed). Another ipad was able to open the files. I have tried to open it in android both via bluestacks emulator and natively on the latest android and an old android version. Here they are:

[troubled files.zip](https://github.com/LLK/scratchjr/files/6176151/troubled.files.zip)

Project 33 opens, project 32 doesn't. They were created on the same, fully updated (I believe) ipad within a few days of each other.

There is another problem, possibly related, which is demonstrated in [this video](https://i.imgur.com/PuRiGsS.mp4).
